frontend/src/main/web/lib/components/SystemGlossary.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import Configs from '../constants/Configs';
+import GlossaryStore from '../stores/GlossaryStore';
+import { PureRenderMixin } from 'react/addons';
+import Actions from '../actions/GlossaryActions';
+import { Button, Input, Icons, Icon, Select } from 'zanata-ui'
+import DataTable from './glossary/DataTable'
+import NewEntryModal from './glossary/NewEntryModal'
+import ImportModal from './glossary/ImportModal'
+import { Loader } from 'zanata-ui'
+import _ from 'lodash';
+import StringUtils from '../utils/StringUtils'
+
+var SystemGlossary = React.createClass({
+  mixins: [PureRenderMixin],
+
+  _init: function() {
+    return GlossaryStore.init();
+  },
+
+  getInitialState: function() {
+    return this._init();
+  },
+
+  componentDidMount: function() {
+    GlossaryStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    GlossaryStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    this.setState(this._init());
+  },
+
+  _handleTransChange: function(localeId) {
+    Actions.changeTransLocale(localeId)
+  },
+
+  _handleFilterKeyDown: function(event) {
+    if(event.key == 'Enter') {
+      Actions.updateFilter(this.state.filter);
+    }
+  },
+
+  _handleFilterChange: function(event) {
+    this.setState({filter: event.target.value});
+  },
+
+  render: function() {
+    var count = 0,
+      selectedTransLocale = this.state.selectedTransLocale,
+      uploadSection = null, newEntrySection = null;
+
+    var contents = (<DataTable
+      glossaryData={this.state.glossary}
+      glossaryResId={this.state.glossaryResId}
+      focusedRow={this.state.focusedRow}
+      hoveredRow={this.state.hoveredRow}
+      totalCount={this.state.glossaryResId.length}
+      canAddNewEntry={this.state.canAddNewEntry}
+      canUpdateEntry={this.state.canUpdateEntry}
```

Comment:
Lots of these have the same name in state and DataTable props. Consider using spread attributes, and maybe `_.pick`